Sources/StreamVideo/WebRTC/StreamVideoCaptureHandler.swift

```diff
@@ -34,21 +34,28 @@ class StreamVideoCaptureHandler: NSObject, RTCVideoCapturerDelegate {
     
     func capturer(_ capturer: RTCVideoCapturer, didCapture frame: RTCVideoFrame) {
         guard let buffer: RTCCVPixelBuffer = frame.buffer as? RTCCVPixelBuffer else { return }
-        Task {
-            let imageBuffer = buffer.pixelBuffer
-            CVPixelBufferLockBaseAddress(imageBuffer, .readOnly)
-            let inputImage = CIImage(cvPixelBuffer: imageBuffer, options: [CIImageOption.colorSpace: colorSpace])
-            let outputImage = await filter(image: inputImage)
-            CVPixelBufferUnlockBaseAddress(imageBuffer, .readOnly)
-            self.context.render(outputImage, to: imageBuffer, bounds: outputImage.extent, colorSpace: colorSpace)
-            let updatedFrame = handleRotation ? adjustRotation(capturer, for: buffer, frame: frame) : frame
+        Task { [weak self] in
+            guard let self else { return }
+            await applyFilter(on: buffer)
```

Comment:
I haven't :) It will be great if you can share some ways to test this one as i can't find any filters in the app

Reply:
We have guide here: https://getstream.io/video/docs/ios/advanced/apply-video-filters/. We also have it in the dogfooding app.
Orientation is a bit more tricky, since there were issues on older iPads, so we would need to do a lot of testing for this change. That's why I wouldn't do any rotation changes, since that stuff is tested and works.
We can just prevent the buffer stuff directly in the method.